spec/factories/edition_factory.rb

```diff
@@ -102,6 +106,10 @@
           revision_at_creation: edition.revision,
           withdrawal: evaluator.withdrawal,
         )
+        unless edition.document.first_published_at
+          edition.document.first_published_at = Time.current.yesterday
+          edition.document.save!
+        end
```

Comment:
I'm keen to see if we can do the factories in a different way to avoid the use of a save method (as this will mean that if we build an edition it will persist a document but none of the other models). It also wouldn't work correctly if the status isn't persisted.

I had a quick go myself at trying to put something together that's consistent with our other transient attributes to see if we could get around this requirement: https://github.com/alphagov/content-publisher/compare/first_published_at-factory?expand=1

Reply:
This is great - I had tried a similar approach but missed out adding `first_published_at` in the transient block, so it didn't work.  Updated 👍